[WebGPU] RenderPassEncoder::beginOcclusionQuery will crash when offset%8 != 0

mwyrzykowski opened the pull request:
#### 2145402eba85d243bfe2bfd6009dea8f341cc270
<pre>
[WebGPU] RenderPassEncoder::beginOcclusionQuery will crash when offset%8 != 0
<a href="https://bugs.webkit.org/show_bug.cgi?id=254177">https://bugs.webkit.org/show_bug.cgi?id=254177</a>
&lt;radar://106959317&gt;

Reviewed by Myles C. Maxfield.

In QuerySet.mm, Device::createQuerySet already creates a buffer
with size that is a multiple of 8.

-[MTLRenderCommandEncoder setVisibilityResultMode:offset:] requires
offset to be a multiple of 8. It is the offset into the buffer.

queryIndex is the index of the visiblity query. Each index is offset
8 bytes in the buffer.

So we can simply multiply the index by 8 here.

* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::beginOcclusionQuery):

Canonical link: <a href="https://commits.webkit.org/261934@main">https://commits.webkit.org/261934@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/42fa7e7b8eeb1fd3c5e0d025805a03c5f78b5859

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/113293 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/22426 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1961 "Built successfully") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121746 "Built successfully") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23787 "Built successfully") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/119079 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100940 "Build is in progress. Recent messages:OS: Monterey (12.6.3), Xcode: 14.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-api-tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/106389 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/1480 "Build is in progress. Recent messages:OS: Monterey (12.6.3), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; Running layout-tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/46730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14726 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/1519 "Build is in progress. Recent messages:OS: Monterey (12.6.3), Xcode: 14.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; Running layout-tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/98985 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/15434 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20728 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/53510 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8343 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/17289 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

djg reviewed (commented):
Is 8 just a magic number or the size of uint64_t, struct, etc? If it is, can we use size of the correct type instead?

djg reviewed (approved):
:shipit:

litherum reviewed (approved):
Why no test?

mwyrzykowski committed:
[WebGPU] RenderPassEncoder::beginOcclusionQuery will crash when offset%8 != 0
https://bugs.webkit.org/show_bug.cgi?id=254177
<radar://106959317>

Reviewed by Myles C. Maxfield.

In QuerySet.mm, Device::createQuerySet already creates a buffer
with size that is a multiple of 8.

-[MTLRenderCommandEncoder setVisibilityResultMode:offset:] requires
offset to be a multiple of 8. It is the offset into the buffer.

queryIndex is the index of the visiblity query. Each index is offset
8 bytes in the buffer.

So we can simply multiply the index by 8 here.

* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::beginOcclusionQuery):

Canonical link: https://commits.webkit.org/261934@main
